### DISCUSSION TITLE
Remember GPX waypoint overlays in import route

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#46] [Q&A] [answered]
Be able to generate multiple navigation tracks into a single gpx file to share is also a  nice to have.
Know that in the gpx standard the gpx waypoints only do represent free located individual favorite points.
Create free located (poi) wpts and **merge them together** with the exported (nav)tracks into a single gpx file.

**Comment by devemux86** [comment]
GPX navigation track is for storing routes ready for offline navigation.
We better not mix it with extra data that will be complicated to handle.

**Comment by devemux86** [comment]
@0709wiwiwi
I am not sure what `wpt` do you expect to see in a GPX with a navigation track inside?
The route's waypoints or some random unrelated favorites?

**Reply by 0709wiwiwi** [reply]
Exactly. The gpx wpt that do not participate in the construction of a track but are present and transfered inclusive a (rte)route and or a (trk) track in the same gpx file. As indicated in the gpx standard and also as used in klm files.

**Reply by devemux86** [reply]
@0709wiwiwi So you mean to include in the nav gpx the route's waypoints as `wpt`?

**Reply by devemux86** [reply]
During the export of routes, the GPX is one of the formats to contain the routes.
(as nav or normal `trk` or `rte` or `wpt`)

To export favorites, can use the favorites screen and export them in GPX format.
To import favorites, use the favoties screen and select icon, color, visibility, etc.

**Reply by devemux86** [reply]
Also `sym` is used in nav tracks, so it is not available for icons.

**Reply by 0709wiwiwi** [reply]
Don't make that detour with those complex favorites. Favorites lists are not always user-friendly and or quickly to edit or to use.
Update: Simple as in the next video: [gpx track add single gpx waypoint](https://youtube.com/shorts/l4xXpqQxSeE)
Exported file:  [2_Locus_trk_wpt.gpx.txt](https://github.com/user-attachments/files/18695296/2_Locus_trk_wpt.gpx.txt)

**Reply by devemux86** [reply]
> edit

Edit? Users do not use editors, they use apps and websites to calculate routes and generate GPX files.

So the suggestion is during import route, like we can now store permanently the imported track overlay in my collection,
to also store permanently the imported GPX wpt as favorites in my collection?

The thread title mentions "export", so it is a bit confusing about what really is the suggestion
and what actual app steps (without editors) are expected users to do.

**Reply by MotoUKRider** [reply]
@0709wiwiwi I'm not 100% certain what level of detail is being requested here, and something is being lost in translation as some sentences do not make sense (to me, at least!).

Whatever workflow you are trying to achieve, it's not something I know anything about.

**Reply by devemux86** [reply]
@0709wiwiwi You want to export a route + favorites in the same GPX file?
Route as nav `trk` and favorites as `wpt`?

What favorites? How to select what to include in the GPX?

**Reply by devemux86** [reply]
@0709wiwiwi Please answer our questions so we can understand what you mean:

> You want to export a route + favorites in the same GPX file?
Route as nav trk and favorites as wpt?
What favorites? How to select what to include in the GPX?

**Reply by devemux86** [reply]
> add an extra third waypoint inclusive sym that represent the Garmin Bar Icon

> With Locus identical and super simple to create such combination as ever possible.

How? With what interface?
We may have too many favorites on the map. How to include only 2-3 of them in the GPX?

**Reply by 0709wiwiwi** [reply]
> They use apps and websites to calculate routes and generate GPX files.

Like Garmin Mapsource - Basecamp. 
Possibly somewhat outdated, but anyway stable, and still a great reference. 
Locus and several other apps generate these just as similar indeed. 
Indeed.

**Comment by devemux86** [comment]
Let's organize the discussion based on the real use with the app interface:

- **Import route** from GPX
We could import `wpt` in favorites, so they are remembered (like track overlays).

If users are interested in this, I could look into it.

**Reply by devemux86** [reply]
Are there any comments here?
Do you find these additions to Import / Export useful?

**Comment by 0709wiwiwi** [comment]
**Colorful gpx waypoint icons.**

The gpx standard indicates favorites locations as simple gpx waypoints. 
They are often just temporarily necessary not to be stored in favorites.
These gpx waypoints can accompany a (disposable) (nav)track design.

DEMO. 
- Import Cruiser wpt sym sports_bar and (nav)track.
[2_GPX_wpt_color_cruiser_sym_icon_navtrk.gpx.txt](https://github.com/user-attachments/files/24099235/2_GPX_wpt_color_cruiser_sym_icon_navtrk.gpx.txt)

Current display. 
Cruiser shows the overlay waypoints as **color** dots.
![wpt_color](https://github.com/user-attachments/assets/cf2de3c3-bb17-446a-bb07-82528e5405e4)

Expected display.
 Cruiser displays the Cruiser color **sports_bar icons.**
<img width="350" height="203" alt="Expected_display" src="https://github.com/user-attachments/assets/8a4f39d2-1dec-481f-9894-3bad4a777f47" />

**Comment by devemux86** [comment]
In next version when import a GPX as overlay with waypoints and track:
- GPX track will be stored in group "Imported tracls" in "My collection | Tracks".
GPX track `name`, `desc` and `color` (via extension) will be included.

- GPX waypoints will be stored in group "Imported favorites" in "My collection | Favorites".
GPX waypoint `name`, `desc`, `sym` (if compatible) and `color` (via extension) will be included.

- "Routing | Clear | Overlay" action will clear both of them automatically.

**Reply by bios4** [reply]
Will this be done always automatically or will there be a switch in settings if these items should be stored to My collection?

This feature confuses me, and to be honest I have no idea why I would need all these items stored automatically...

What's the benefit of having lots and lots of single wpts added in favorites?

**Reply by devemux86** [reply]
@bios4

"Routing | Import" as overlay already stores the track in my collection in any 1st group.
- https://github.com/devemux86/cruiser/discussions/431

What is better organized that there will be a dedicated group for the imported overlays.

**Reply by bios4** [reply]
I understand the storing of the track, but am still confused why I would want to have the single wpts in the favorites...

**Reply by bios4** [reply]
> "Routing | Clear | Overlay" action will clear them automatically.

Are the wpts deleted also when a new GPX is imported over the first one without executing the "Clear" operation?

**Reply by devemux86** [reply]
@bios4 

Users asked to remember the overlays during import.
Overlays can be tracks and/or waypoints.

They are deleted like now. Manually with the "Clear" operation.
(or in my collection)

**Comment by devemux86** [comment]
If we do not need to remember the waypoint overlays during import (like we already remember the track overlays)
then we can skip this feature and should mark this thread as resolved because there is nothing to implement here.

**Comment by 0709wiwiwi** [comment]
> merge them **together**

Overlay gpx track(s) with poi waypoint(s) stay together.
They come together, show up together, and leave together.
Nothing more nothing less. Unclear ? Then close.

**Comment by devemux86** [comment]
@0709wiwiwi Please do not close threads where there is a discussion with users.

**Reply by 0709wiwiwi** [reply]
Since we keep circling the pot. Close.

**Reply by devemux86** [reply]
@0709wiwiwi Better to mark a post as an answer to be easily found in the future.

**Reply by 0709wiwiwi** [reply]
Programs and apps examples were mentioned but not used. Thus, no good answer is selectable. 
It is in danger of going completely wrong so better is to stop this now so no damage is done.

**Comment by MotoUKRider** [comment]
This topic seemed like a good place to try...  I very rarely use Tracks but if I pilfer a GPX from a bike group or website, it'll usually contain a Track which is very helpful to me as I can build a route in BaseCamp; using it as an overlay to iron-out any routing defaults I have.

The example here is the Wainwright Coast-to-Coast which is a hiker's route from one side of England to the other.  I intend to ride via motorcycle so I have worked out a route that follows it as closely as possible (Basecamp image below).

_Problem is, I can't work out how to display *both* in Cruiser App._

I can add+navigate my Bike Route GPX (purple) like any other.  I can add the original "Walk" GPX (attached, and shown in dark grey) as an Overlay but can't do anything else with it, ie: save it into my "Tracks" Collection.  It won't import from inside Collection>Tracks - which I'd expect to work as it's a "no calculate" import (like Favorites).

My custom motorcycle Route has 21 routing points whereas the Track has 2114 which is entirely expected - I was under the impression that I could load a Track, save it into Collections and toggle it as an Overlay, adding a Route to also display if I so desired.  It's a nice grey dashed line in Cruiser App, but I can't "touch" it to do anything more with.

Please note that I'm wearing my _Stupid User_ hat here, and hopefully trying to achieve something that somebody else might want to do but struggle with!

Wainwright Walk Track: [wainwright-coastal-walk.txt](https://github.com/user-attachments/files/23849593/wainwright-coastal-walk.txt)

My BaseCamp view for info:
![wainwright-routes](https://github.com/user-attachments/assets/fe314437-0450-4d6b-bc4a-ca6390148dbe)

**Reply by devemux86** [reply]
@MotoUKRider The GPX file does not contain any tracks `<trk>`.

It contains only a route `<rte>`, not to be confused with tracks.

**Reply by devemux86** [reply]
@MotoUKRider Try exporting the GPX as a real track with `<trk>`.

**Reply by 0709wiwiwi** [reply]
In Basecamp set + export as track.
[wainwright-coastal-walk.gpx.txt](https://github.com/user-attachments/files/23850660/wainwright-coastal-walk.gpx.txt)

**Reply by MotoUKRider** [reply]
@devemux86 It's Monday morning and I've only had 1 cup of coffee - and that _Stupid User_ hat fits better than I thought!

Look what I made...
![cruiser-route-track-overlay](https://github.com/user-attachments/assets/559a45aa-b916-4434-8b25-f5edfe3cfcd1)

Got the Track imported into Collections, and also loaded the Route for nav - perfect!
[wainwright-coastal-walk.txt](https://github.com/user-attachments/files/23850774/wainwright-coastal-walk.txt)

That "Track" in BaseCamp was actually a Route - and I should have known this because the icons for them are the same - so after converting it to an actual Track, everything works.  Routes in BaseCamp have the green square icons (plus an image overlay for the Profile, eg: Motorcycle).  Actual Tracks have the footprints icon.

Here's what I now have - Route, Route from Track (about to be deleted), Track:
![basecamp-item-types](https://github.com/user-attachments/assets/69a1ad27-8b3f-4248-84d3-6462f1b77a0d)

![basecamp-route-track-view](https://github.com/user-attachments/assets/6023252c-bb8f-48c2-9336-5f23eb02ac4b)

Luckily there's rain all day today, as I don't think I should be on the road in this state :clown_face:

**Comment by devemux86** [comment, marked as answer]
- Implemented in https://github.com/devemux86/cruiser/discussions/891
